### ISSUE TITLE
Enhance splitting function

### ISSUE THREAD

**Issue by saiboxx** [slds-lmu/wildlife-ml#4]
Currently the splitting function `do_train_split` in `data.py` in the `preprocessing` branch does only support a randomized split in train and test set.

Tasks:
- Extend the function to allow `stratified` splitting.
- Extend the function to allow splitting in train, test and validation sets.
